### PR TITLE
feature/blui-4366-empty_state-RNPv5

### PR DESCRIPTION
**Pull request by ArshdeepSingh3-eaton**
Changes proposed in this Pull Request:
- Updated the empty state component

 Screenshots / Screen Recording (if applicable)

Screenshot 2023-09-18 at 1 30 17 PM
Screenshot 2023-09-18 at 1 40 56 PM

To Test:

for ios
- cd MD3
- yarn ios
for android
-yarn android

Any specific feedback you are looking for?
-

### CODE REVIEW

**Comment by daileytj** [comment]
This is probably an oversight on my communication but can you move this component into the `src/md3` folder and work out of the showcase demo in the `demos/showcase` folder instead of the MD3 app.

File: MD3/components/empty-state/empty-state.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            padding: 16,
+        },
+        title: {
+            textAlign: 'center',
+            marginTop: 16 * fontScale,
+            fontSize: 20, // Customize as needed
+            letterSpacing: 0, // Customize as needed
+        },
+        description: {
+            //   color: theme.colors.primary,
```
Comment (daileytj) [inline MD3/components/empty-state/empty-state.tsx:31]: If we're not using this let's get rid of it

File: MD3/components/empty-state/empty-state.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        root: {
+            flex: 1,
+            justifyContent: 'center',
+            alignItems: 'center',
+            padding: 16,
+        },
+        title: {
+            textAlign: 'center',
+            marginTop: 16 * fontScale,
+            fontSize: 20, // Customize as needed
```
Comment (daileytj) [inline MD3/components/empty-state/empty-state.tsx:27]:
```suggestion
            fontSize: 22, // Customize as needed
```

File: MD3/components/empty-state/empty-state.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        if (icon) {
+            return <Icon source={icon} size={normalizeIconSize()} color={getColor(iconColor)} />;
+        }
+    }, [icon, normalizeIconSize, getColor, iconColor]);
+
+    return (
+        <View style={[defaultStyles.root, styles.root, style]} {...viewProps}>
+            {getIcon()}
+            <Text style={[defaultStyles.title, styles.title]}>{title}</Text>
+            {description ? <Text style={[defaultStyles.description, styles.description]}>{description}</Text> : null}
```
Comment (daileytj) [inline MD3/components/empty-state/empty-state.tsx:91]: We can probably let the variants handle the font size in the long term. For now lets add the variants but leave the styles in place until theming is finished in the showcase.

```suggestion
            <Text style={[defaultStyles.title, styles.title]} variant={'titleLarge'}>{title}</Text>
            {description ? <Text style={[defaultStyles.description, styles.description]} variant={'bodyMedium'}>{description}</Text> : null}
```

**Comment by huayunh** [comment]
why does the font color look different between Android and iOS?

(The letter spacing of list item tag on iOS look different from Android's too)

File: components/src/md3/empty-state/empty-state.tsx [hunk 4]
```diff
@@ -76,7 +77,7 @@ export const EmptyState: React.FC<EmptyStateProps> = (props :EmptyStateProps) =>
         return Math.max(100, Math.min(200, iconSize));
     }, [iconSize]);
 
-    const getColor = useCallback((color: string | undefined): string => color || theme.colors.primary, [theme]);
+    const getColor = useCallback((color: string | undefined): string => color || theme.colors.outlineVariant, [theme]);
```
Comment (daileytj) [inline components/src/md3/empty-state/empty-state.tsx:80]: We may want to leave a `// @TODO:` comment here mentioning that we will need to update this to theme.colors.disabled long term

**Review by daileytj — APPROVED** [review]

**Pull request merged**